onelogin_aws_cli/cli.py

```diff
@@ -48,6 +50,11 @@ def login(args=sys.argv[1:]):
     api.save_credentials()
 
     if args.renewSeconds:
-        while True:
-            time.sleep(args.renewSeconds)
-            api.save_credentials()
+        refresh_process = ForegroundProcess(period=args.renewSeconds, api=api)
```

Comment:
Something like

```
while not self._sleep.is_set():
  self._sleep.wait(self.period)
  api.save_credentials()
```

And the sig handler part does:

```
signal.signal(signal.SIGINT, lambda: self._sleep.set())
```

Reply:
Ah, ok, lemme look through the other thread and internalize that first =).

Reply:
I just added some ideas from head to expand on point 2 in #36.

Reply:
And I see what you mean now, yes that would work fine for the reasoning I gave above. :-) I might just do that anyway, as it solves the problem I mentioned and do the Thread class at a later date.